

<launch>
  <arg name="launch_driver" default="true"/>
  <arg name="interface" default="can0"/>
  <arg name="receiver_interval_sec" default="0.01"/>

  <arg name="sensor_model" default="aip_x1"/>
  <let name="vehicle_id" value="$(env VEHICLE_ID default)"/>
  <let name="individual_params_config" value="$(find-pkg-share individual_params)/config/$(var vehicle_id)/$(var sensor_model)"/>

  <group>
    <push-ros-namespace namespace="imu"/>
    <group>
      <push-ros-namespace namespace="tamagawa"/>
      <include file="$(find-pkg-share ros2_socketcan)/launch/socket_can_receiver.launch.py">
        <arg name="interface" value="$(var interface)"/>
        <arg name="interval_sec" value="$(var receiver_interval_sec)"/>
      </include>
      <node pkg="tamagawa_imu_driver" name="tag_can_driver" exec="tag_can_driver" if="$(var launch_driver)">
        <remap from="/can/imu" to="from_can_bus"/>
        <remap from="/imu/data_raw" to="imu_raw"/>
        <param name="imu_frame_id" value="imu_link"/>
      </node>
      <include file="$(find-pkg-share imu_corrector)/launch/imu_corrector.launch.xml">
      <arg name="input_topic" value="imu_raw"/>
      <arg name="output_topic" value="imu_data"/>
      <arg name="param_file" value="$(var individual_params_config)/imu_corrector.param.yaml"/>
      </include>
    </group>
  </group>
</launch>